src/rubocop/Rubocop.js

```diff
@@ -0,0 +1,83 @@
+'use babel'
+
+import pluralize from 'pluralize'
+import parseFromStd from '../helpers/std-parser'
+import RubocopRunner from './RubocopRunner'
+import ErrorFormatter from '../ErrorFormatter'
+import OffenseFormatter from './OffenseFormatter'
+
+const PARSE_ERROR_MSG = 'Rubocop: Parse error'
+const UNEXPECTED_ERROR_MSG = 'Rubocop: Unexpected error'
+const UNDEF_VERSION_ERROR_MSG = 'Unable to get rubocop version from linting output results.'
+const NO_FIXES_INFO_MSG = 'Linter-Rubocop: No fixes were made'
+
+export default class Rubocop {
+  constructor(config) {
+    this.config = config
+  }
+
+  async autocorrect(filePath) {
+    if (!filePath) { return }
+
+    try {
+      const output = await new RubocopRunner(this.config).run(filePath, ['--auto-correct', filePath])
+      try {
+        // Process was canceled by newer process or there is nothing to parse
+        if (output === null) { return }
```

Comment:
🚨 An autocorrect run should be blocking on all `lint` calls and shouldn't be able to be interrupted by a lint call!

Reply:
Related to this is, How to block a lint call on `editor.onDidSave()`?, I am thinking in implement a `Fix on save` option but the `lint` call blocks the `autocorrect` call.